config/main.py

```diff
@@ -1777,7 +1778,15 @@ def vrf_add_management_vrf(config_db):
     config_db.mod_entry('MGMT_VRF_CONFIG', "vrf_global", {"mgmtVrfEnabled": "true"})
     mvrf_restart_services()
 
-def vrf_delete_management_vrf(config_db):
+    cmd = "cat /proc/net/route | grep -E \"eth0\s+00000000\s+[0-9A-Z]+\s+[0-9]+\s+[0-9]+\s+[0-9]+\s+202\" | wc -l"
```

Comment:
>grep [](start = 33, length = 4)

Could you explain the regex?

Reply:
The output for kernel route will be in the following format .
	root@sonic:~# cat /proc/net/route
	Iface   Destination     Gateway         Flags   RefCnt  Use     Metric  Mask            MTU     Window  IRTT                                                 
	eth0    00000000        01803B0A        0003    0       0       202     00000000        0       0       0                                                    
The regular expression does search string for fields Iface   , Destination  , Gateway  , Flags   , RefCnt  , Use  and 
Metric . Here for the above entry :
Iface: eth0
Destination  : 00000000        
Gateway  : [0-9A-Z]
Flags   : [0-9]
RefCnt  : [0-9]
Use   : [0-9]
Metric : 202

Reply:
For the sake of future reader, add some comments
```python
'''
To match eth0 line in /proc/net/route, sample file:
$ cat /proc/net/route
Iface Destination Gateway Flags RefCnt Use Metric Mask MTU Window IRTT
eth0 00000000 01803B0A 0003 0 0 202 00000000 0 0 0
...
'''
```

---
In reply to: [536375136](https://github.com/Azure/sonic-utilities/pull/1289#discussion_r536375136) [](ancestors = 536375136)

Reply:
done , submitted in patch